test/anchoredinterval.jl

```diff
@@ -107,8 +107,27 @@ using Intervals: Bounded, Ending, Beginning, canonicalize, isunbounded
 
         he = HourEnding(dt)
         hb = HourBeginning(dt)
-        @test_throws DomainError convert(DateTime, he)
-        @test_throws DomainError convert(DateTime, hb)
+
+        # Note: replace try/catch with the following when deprecation is dropped
+        # @test_throws DomainError convert(DateTime, he)
+        @test_broken try
```

Comment:
Is `@test_broken` the right thing to use here? Is this expected behavior for this version (that will soon change) or just a bug that nobody noticed?

Reply:
I think this is definitely debatable. Specifically the tests which expect a `DomainError` are broken. However, for this version we do expect that the convert will pass so I could use a `@test` here. When the deprecation is dropped that test will fail and result in the tests also being updated which is really what I'm after.

Reply:
I'll change this as the current implementation is probably obfuscating things

Reply:
You could just use `@test_deprecated` which will fail when the code changes